File: pset1/greedy.c
#include <cs50.h>
#include <stdio.h>

    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;
    float dollars;
    int change;
    int counter;
int main(void)
{
    do
    {
        printf("O hail! How much change is owed?\n");
        dollars = get_float(); 
    }
    while (change < 0.00);
    
    change = dollars * 100;
    
    while (change - quarter >= 0)
    {
        counter++;
        change = change - quarter;
        printf("Quarters change: %i\n",change);
    }
    while (change - dime >= 0)
    {
        counter++;
        change = change - dime;
        printf("Dime change: %i\n",change);
    }
    while (change - nickel >= 0)
    {
        counter++;
        change = change - nickel;
        printf("Nickel change: %i\n",change);
    }
        while (change - penny >= 0)
    {
        counter++;
        change = change - penny;
        printf("Penny change:%i\n",change);
    }
    

    printf("Rest: %i \n", change);
    printf("Coins change:%i \n", counter);
}

File: pset6/greedy.py
import cs50

quarter = 25.00
dime = 10.00
nickel = 5.00
penny = 1.00
dollars = 0.00
change = 0.00
counter = 0

while True:
    print("O hail! How much change is owed? ", end="")
    dollars = cs50.get_float()
    if dollars > 0.00:
        break

change = dollars * 100.00

counter += change // quarter
change %= quarter
print("Quarters ($.25) change: ", round(change/100,2), "Coins change: ", round(counter))

if change != 0:
    counter += change // dime
    change %= dime
    print("Dime ($.10) change: ",  round(change/100,2), "Coins change: ", round(counter))
    
    if change != 0:
        counter += change // nickel
        change %= nickel
        print("Nickel ($.05) change: ", round(change/100,2), "Coins change: ", round(counter))
       
        if change != 0:
            counter += change // penny
            change %= penny
            print ("Penny ($.01) change: ", round(change/100,2), "Coins change: ", round(counter))

print()    
print("Rest: ", round(change,2))
print("Coins change: ", round(counter))

 
    
